### PR TITLE
Documentation workflow: auto-publish from master to gh-pages

### PR DESCRIPTION
**Pull request by Guts**
Documentation is stored in a separate branch, which is the published one as static website on GitHub Pages.

The goal of this PR is to store the documentation into the `docs` folder and to automatically generate and deploy to gh-pages through a GitHub Action.

Advantages:

- embed other files (changelog, main readme...)
- encourage to update the docs during development

### CODE REVIEW

**Review by havatv — APPROVED** [review]
Looks fine

**Review by havatv — COMMENTED** [review]
I trust your judgement
.Feeel free to merge.

**Comment by Guts** [comment]
New doc is ready. For safety, I've duplicated existing gh-pages content to https://github.com/QGIS-Contribution/QGIS-ResourceSharing/tree/gh-pages-backup.

Here is a snapshot of how the new doc looks:

image

File: docs/authoring/creating-metadata.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+The metadata file (`metadata.ini`) must have a `general` section with this information:
+
+| Name | Status | Description |
+| :--- | :----: | :---------- |
+| collection | Required | Comma separated list of collection names (these have to match the directory names of the collections) |
+
+## Collection
+
+For each collection, metadata are needed. The section must be named by the name defined in the general section.
```
Comment (havatv) [inline docs/authoring/creating-metadata.md:15]: by the name"-> using the name"?

**Review by havatv — COMMENTED** [review]
Sorry for being late.
Looks fine. I 8have one suggestion.

**Pull request merged**